compiler/rustc_typeck/src/check/intrinsic.rs

```diff
@@ -107,14 +112,24 @@ pub fn intrinsic_operation_unsafety(intrinsic: Symbol) -> hir::Unsafety {
         | sym::variant_count
         | sym::ptr_mask => hir::Unsafety::Normal,
         _ => hir::Unsafety::Unsafe,
-    }
+    };
+
+    assert_eq!(
+        has_safe_attr,
+        is_in_list,
+        "Intrinsic safety mismatch between list of intrinsics within the compiler and core library intrinsics for intrinsic `{}`",
+        tcx.item_name(intrinsic_id)
+    );
```

Comment:
I think instead of ICE-ing the compiler, it would be better to report this as an error to the user. You can do that by calling [`tcx.sess.err()`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_session/session/struct.Session.html#method.err).

Reply:
Sounds good - I was debating between an ICE and a regular error, and opted for an ICE as from the Zulip discussion it seems to be more of an implementation/compiler error than a user error. But a regular error makes more sense for people reimplementing the core library. Done in 062b1035698